File: client/app/src/main/java/com/rh/heji/MainViewModel.java
package com.rh.heji;

import androidx.lifecycle.ViewModel;

/**
 * Date: 2020/11/3
 * Author: 锅得铁
 * #
 */
public class MainViewModel extends ViewModel {
    /**
     * 控制toolbar
     */
    String homeUUID;
    String reportUUID;
    String settingUUID;

    public String getHomeUUID() {
        return homeUUID;
    }

    public void setHomeUUID(String homeUUID) {
        this.homeUUID = homeUUID;
    }

    public String getReportUUID() {
        return reportUUID;
    }

    public void setReportUUID(String reportUUID) {
        this.reportUUID = reportUUID;
    }

    public String getSettingUUID() {
        return settingUUID;
    }

    public void setSettingUUID(String settingUUID) {
        this.settingUUID = settingUUID;
    }
}

File: client/app/src/main/java/com/rh/heji/MainViewModel.kt
package com.rh.heji

import androidx.lifecycle.MediatorLiveData
import androidx.lifecycle.ViewModel
import com.rh.heji.data.db.Image
import com.rh.heji.data.repository.BillRepository
import com.rh.heji.data.repository.CategoryRepository
import com.rh.heji.utlis.YearMonth
import java.util.*

/**
 * @date: 2020/11/3
 * @author: 锅得铁
 * # APP运行时 UI常量共享存储
 */
class MainViewModel : ViewModel() {
    var billRepository = BillRepository()
    var categoryRepository = CategoryRepository()

    private val imageLiveData = MediatorLiveData<MutableList<Image>>()

    /**
     * 全局选择的年月（home to subpage）
     */
    var globalYearMonth: YearMonth =
        YearMonth(Calendar.getInstance()[Calendar.YEAR], Calendar.getInstance()[Calendar.MONTH] + 1)

    /**
     * 当前年月
     */
    val currentYearMonth by lazy {
        YearMonth(
            Calendar.getInstance()[Calendar.YEAR],
            Calendar.getInstance()[Calendar.MONTH] + 1
        )
    }
}